shell/platform/fuchsia/flutter/pointer_injector_delegate.cc

```diff
@@ -0,0 +1,318 @@
+
+namespace flutter_runner {
+
+using fup_Config = fuchsia::ui::pointerinjector::Config;
+using fup_Context = fuchsia::ui::pointerinjector::Context;
+using fup_Data = fuchsia::ui::pointerinjector::Data;
+using fup_DeviceType = fuchsia::ui::pointerinjector::DeviceType;
+using fup_DispatchPolicy = fuchsia::ui::pointerinjector::DispatchPolicy;
+using fup_Event = fuchsia::ui::pointerinjector::Event;
+using fup_EventPhase = fuchsia::ui::pointerinjector::EventPhase;
+using fup_PointerSample = fuchsia::ui::pointerinjector::PointerSample;
+using fup_Target = fuchsia::ui::pointerinjector::Target;
+using fup_Viewport = fuchsia::ui::pointerinjector::Viewport;
+using fuv_ViewRef = fuchsia::ui::views::ViewRef;
+const auto fup_MAX_INJECT = fuchsia::ui::pointerinjector::MAX_INJECT;
+
+namespace {
+
+// clang-format off
+  static constexpr std::array<float, 9> kIdentityMatrix = {
+    1, 0, 0, // column one
+    0, 1, 0, // column two
+    0, 0, 1, // column three
+  };
+// clang-format on
+
+}  // namespace
+
+bool PointerInjectorDelegate::HandlePlatformMessage(
+    rapidjson::Value request,
+    fml::RefPtr<flutter::PlatformMessageResponse> response) {
+  if (!registry_->is_bound()) {
+    FML_LOG(WARNING)
+        << "Lost connection to fuchsia.ui.pointerinjector.Registry";
+    return false;
+  }
+
+  auto method = request.FindMember("method");
+  if (method == request.MemberEnd() || !method->value.IsString()) {
+    return false;
+  }
+
+  if (method->value == "View.Pointerinjector.inject") {
+    auto args_it = request.FindMember("args");
+    if (args_it == request.MemberEnd() || !args_it->value.IsObject()) {
+      FML_LOG(ERROR) << "No arguments found.";
+      return false;
+    }
+
+    const auto& args = args_it->value;
+
+    auto view_id = args.FindMember("viewId");
+    if (!view_id->value.IsUint64()) {
+      FML_LOG(ERROR) << "Argument 'viewId' is not a uint64";
+      return false;
+    }
+    auto id = view_id->value.GetUint64();
+
+    auto phase = args.FindMember("phase");
+    if (!phase->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'phase' is not a int";
+      return false;
+    }
+
+    auto pointer_x = args.FindMember("x");
+    if (!pointer_x->value.IsFloat() && !pointer_x->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'Pointer.X' is not a float";
+      return false;
+    }
+
+    auto pointer_y = args.FindMember("y");
+    if (!pointer_y->value.IsFloat() && !pointer_y->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'Pointer.Y' is not a float";
+      return false;
+    }
+
+    auto pointer_id = args.FindMember("pointerId");
+    if (!pointer_id->value.IsUint()) {
+      FML_LOG(ERROR) << "Argument 'pointerId' is not a uint32";
+      return false;
+    }
+
+    auto trace_flow_id = args.FindMember("traceFlowId");
+    if (!trace_flow_id->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'traceFlowId' is not a int";
+      return false;
+    }
+
+    auto view_ref = args.FindMember("viewRef");
```

Comment:
Instead of providing ViewRef here (because its impossible under Flatland), we must harvest it in other ways.

For gfx, provide the ViewRef as an (optional) parameter to View.create
For flatland, wait on the the ChildViewWatcher.GetViewRef return value;  drop pointer events received before that point
you can store them in a std::unordered_map<uint64_t view_id, ViewRef>

Then, remove this required parameter.

Reply:
Done